lib/views/course_view/components/course_card.dart

```diff
@@ -1,144 +1,397 @@
+          Expanded(
+            child: Container(
+              color: themeData.cardColor,
+              padding: const EdgeInsets.all(8.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      _buildCourseTumID(),
+                      _buildCourseIsLive(),
+                    ],
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.symmetric(vertical: 3.0),
+                    child: _buildCourseTitle(themeData.textTheme),
+                  ),
+                  _buildLastLecture(context),
+                ],
+              ),
             ),
           ),
-        ),
+        ],
       ),
     );
   }
 
   Widget _buildCourseImage() {
+    if (path == null) return const SizedBox();
     return Stack(
       children: [
         AspectRatio(
-          aspectRatio: 10 / 7,
+          aspectRatio: 16 / 12, //TODO how are the thumbnails given?
           child: ClipRRect(
             borderRadius: BorderRadius.circular(8.0),
             child: Image.asset(
-              path,
+              path!,
               fit: BoxFit.cover,
             ),
           ),
         ),
+        /*Positioned(
+          bottom: 3, // Adjust this value based on your layout
+          right: 3, // Adjust this value based on your layout
+          child: _buildCourseViewerCount(),
+        ),*/
       ],
     );
   }
 
+  Widget _buildLocation() {
+    if (roomName == null) return const SizedBox();
+
+    final Uri url = roomName != null
+        ? Uri.parse('https://nav.tum.de/room/$roomNumber')
+        : Uri.parse(
+            'https://nav.tum.de/room/',
+          ); //will give an error on NavigaTUM
+
+    return InkWell(
+      onTap: () async {
+        if (await canLaunchUrl(url)) {
+          await launchUrl(url);
+        } else {
+          throw 'Could not launch $url';
+        }
+      },
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          const Icon(
+            Icons.room,
+            size: 20,
+          ),
+          Text(roomName!),
+          //const SizedBox(width: 2),
+          Transform.scale(
+            scale: 0.6, // Adjust the scale factor as needed
+            child: ViewAllButton(onViewAll: () {}), //TODO maybe remove this
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildCourseTumID() {
+    return Text(
+      tumID,
+      overflow: TextOverflow.ellipsis,
+      style: const TextStyle(
+        fontSize: 14,
+        color: Colors.grey,
+        height: 0.9,
+      ),
+    );
+  }
+
   Widget _buildCourseTitle(TextTheme textTheme) {
     return Text(
       title,
       overflow: TextOverflow.ellipsis,
-      maxLines: 3,
+      //maxLines: 2, //TODO check that this never causes overflow
```

Comment:
I am not sure if I understood it correctly, but if you are referring to title text overflow, then maybe you can use sth. like this? https://stackoverflow.com/questions/44579918/flutter-wrap-text-on-overflow-like-insert-ellipsis-or-fade